File: com/bridgelab/dataStructure/CalendarWithStack.java
package com.bridgelab.dataStructure;

import com.bridgeLab.dataStructure.calendar.Calendar;
import com.bridgelab.utility.Queue;
import com.bridgelab.utility.Stack;

public class CalendarWithStack {

	Stack stack1 = new Stack();

	public void calendarWithStack(Queue queue) {
		stack1.push(queue);
	}
	
	public Stack transferFromOneStackToOther() {
		int size = stack1.size();
		Stack stack2 = new Stack();
		for(int i=0; i<size; i++) {
			stack2.push(stack1.peak());
		}
		return stack2;
	}
	
	public static void main(String args[]) {
		Calendar calendar = new Calendar();
		String[][] s1=calendar.calendar(2019, 1);
		String[][] s2=calendar.calendar(2019, 2);
		String[][] s3=calendar.calendar(2019, 3);
		CalendarWithQueue calendarWithQueue = new CalendarWithQueue();
		Queue q1 = calendarWithQueue.weekDayWithQueue(s1);
		Queue q2 = calendarWithQueue.weekDayWithQueue(s2);
		Queue q3 = calendarWithQueue.weekDayWithQueue(s3);
		CalendarWithStack calendarWithStack = new CalendarWithStack();
		calendarWithStack.calendarWithStack(q1);
		calendarWithStack.calendarWithStack(q2);
		calendarWithStack.calendarWithStack(q3);
		Stack stack = calendarWithStack.transferFromOneStackToOther();
		Queue qu1 = (Queue)stack.peak();
		Queue qu2 = (Queue)stack.peak();
		Queue qu3 = (Queue)stack.peak();
		calendarWithQueue.display(qu1);
		calendarWithQueue.display(qu2);
//		calendarWithQueue.display(qu3);

	}
}

File: Month1/DataStructure/CalendarWithStack.py
from DataStructure.Calendar import Calendar
from DataStructure.CalendarWithQueue import CalendarWithQueue


from util.Stack import Stack


class CalendarWithStack:
    def __init__(self):
        """
            Creating two object of stack for storing day of month
        """
        self.stack1 = Stack()
        self.stack2 = Stack()

    def display(self, month, year):
        """
            :param month: take month parameter and display it
            :param year: take year parameter and display it
        """
        calendar = Calendar()
        print("------------", month, year, "---------------")
        for i in calendar.week:
            print(i, end="    ")
        for i in range(self.stack2.size()):
            temp = self.stack2.pop()
            if i % 7 == 0:
                print()
            if int(temp) == 0:
                print(end="     ")
            elif int(temp) < 10:
                print(temp, end="    ")
            else:
                print(temp, end="   ")

    def calendar_with_stack(self, month, year):
        """

            :param m: m as month parameter for calculating month
            :param y: y as year parameter for calculating year
        """
        queue = CalendarWithQueue()
        queue = queue.calendar_with_queue(month, year)
        for i in range(queue.size()):
            self.stack1.push(queue.dequeue())
        for i in range(self.stack1.size()):
            self.stack2.push(self.stack1.pop())


if __name__ == '__main__':
    """
        This is main method
        Creating object of CalendarWithStack class and call its method 
    """
    calendar_with_stack = CalendarWithStack()
    try:
        month = int(input("Enter a month : "))
        if month / 10 > 1.2:
            raise ValueError
    except ValueError:
        print("Month input is invalid.")
    else:
        year = int(input("Enter a year : "))
        calendar_with_stack.calendar_with_stack(month, year)
        calendar_with_stack.display(month, year)
